### PR TITLE
Try shard the golden json file for better diff comparison

### PR DESCRIPTION
**Pull request by shifucun**
When a file is too big, Github refuses to render the diff hence making the golden file diff review impossible.

Try this approach and see if it helps. One possible caveat is when the diff drift to other files, but can try..

### CODE REVIEW

File: test/integration/get_landing_page_data_test.go [hunk 1]
```diff
@@ -95,13 +94,17 @@ func TestGetLandingPageData(t *testing.T) {
 
 		goldenFile := path.Join(goldenPath, c.goldenFile)
 		if generateGolden {
-			updateGolden(result, goldenFile)
+			updateGolden(result, goldenFile, true)
```
Comment (pradh) [inline test/integration/get_landing_page_data_test.go:97]: nit: /* shard= */ true

(useful to know arg in the callsite)

File: test/integration/setup.go [hunk 2]
```diff
@@ -166,15 +170,22 @@ func createBranchTable(ctx context.Context) (*bigtable.Table, error) {
 	return server.SetupBigtable(ctx, data)
 }
 
-func updateGolden(v interface{}, fname string) {
+func updateGolden(v interface{}, fname string, shard ...bool) {
 	jsonByte, _ := json.MarshalIndent(v, "", "  ")
-	err := ioutil.WriteFile(fname, jsonByte, 0644)
+	var err error
+	if len(shard) == 1 && shard[0] {
```
Comment (pradh) [inline test/integration/setup.go:176]: So, is this the way to do default arg in golang?

File: test/integration/setup.go [hunk 3]
```diff
@@ -189,8 +200,65 @@ func updateProtoGolden(resp protoreflect.ProtoMessage, fname string) {
...
+	err := filepath.WalkDir(path, func(s string, d fs.DirEntry, e error) error {
+		if e != nil {
+			return e
+		}
+		if strings.Contains(s, fname) {
+			result = append(result, s)
+		}
+		return nil
+	})
+	return result, err
```
Comment (pradh) [inline test/integration/setup.go:247]: Is result guaranteed to be sorted on name?  Document assumption?

**Review by pradh — APPROVED** [review]
Nice improvement, thanks!

File: test/integration/get_landing_page_data_test.go (hunk 1 above)
Comment (shifucun) [reply test/integration/get_landing_page_data_test.go]: Done

File: test/integration/setup.go (hunk 2 above)
Comment (shifucun) [reply test/integration/setup.go:176]: Yea, it's more of optional args and used as default arg, as there is no native default arg

File: test/integration/setup.go (hunk 3 above)
Comment (shifucun) [reply test/integration/setup.go:248]: Yea, the output is sorted, added comment

**Review by shifucun — COMMENTED** [review]
Thanks for review!

**Pull request merged**